packages/@sanity/cli/src/cli.js

```diff
@@ -24,6 +25,11 @@ module.exports = async function runCli(cliRoot) {
   const cwd = checkCwdPresence()
   const workDir = isInit ? process.cwd() : resolveRootDir(cwd)
 
+  // Try to load .env files from the sanity studio directory
+  // eslint-disable-next-line no-process-env
+  const env = process.env.SANITY_ACTIVE_ENV || process.env.NODE_ENV || 'development'
+  dotenv.config({path: path.join(workDir, `.env.${env}`)})
```

Comment:
What is the use case for having one env file per environment? I'm a bit worried this will encourage keeping `env`-files under source control which is sort of defeating the purpose of env files. It's also advised against as per the official documentation https://github.com/motdotla/dotenv#should-i-have-multiple-env-files. Perhaps we could support it through an explicit option instead?

Reply:
> What is the use case for having one env file per environment? 

Different project IDs/datasets per environment, usually, without having to "remember" the different IDs.
```
// .env.staging
SANITY_STUDIO_API_PROJECT_ID=stagingId
```

> I'm a bit worried this will encourage keeping env-files under source control which is sort of defeating the purpose of env files.

Generally agree, but this concern was not raised back in #1610 - so it's a bit late to change this now, as this is the current and documented behavior: https://www.sanity.io/docs/studio-environment-variables#dot-env-files-env-f5e9e3158896

I'd rather not break this right now - but might be worth changing if we have to do a batch of breaking changes later?

Reply:
Makes a lot of sense. Thanks for the clarification!